client/src/components/category/Categories.js

```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { Route, Switch, useRouteMatch } from "react-router-dom";
+import CategoryForm from "./CategoryForm";
+import CategoriesList from "./CategoriesList";
+
+function Categories() {
+  const { path } = useRouteMatch();
+  return (
+    <Switch>
+      <Route exact path={path}>
+        <CategoriesList />
+      </Route>
+      <Route path={`${path}/new`}>
+        <CategoryForm />
+      </Route>
+      <Route path={`${path}/:id/form`}>
```

Comment:
Why did you rename the route from ../edit to .../form? That breaks the links in the list page...